src/routes/Profile.jsx

```diff
@@ -0,0 +1,193 @@
+// React Imports
+import React, { useContext, useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+// Inrupt Imports
+import { useSession } from '@inrupt/solid-ui-react';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import CheckIcon from '@mui/icons-material/Check';
+import ClearIcon from '@mui/icons-material/Clear';
+import EditIcon from '@mui/icons-material/Edit';
+import Link from '@mui/material/Link';
+import Typography from '@mui/material/Typography';
+// Utility Imports
+import { getBlobFromSolid } from '../utils';
+// Contexts Imports
+import { SignedInUserContext } from '../contexts';
+// Component Inputs
+import { ProfileInputField, ProfileImageField } from '../components/Profile';
+
+/**
+ * Profile Page - Page that displays the user's profile card information and
+ * allow users to edit/update them on PASS
+ *
+ * @memberof Pages
+ * @name Profile
+ * @returns {React.JSX.Element} The Profile Page
+ */
+const Profile = () => {
+  const location = useLocation();
+  const { session } = useSession();
+  const { updateProfileInfo, setProfileData, profileData, fetchProfileInfo } =
+    useContext(SignedInUserContext);
+
+  localStorage.setItem('restorePath', location.pathname);
+
+  const [profileName, setProfileName] = useState(profileData?.profileInfo.profileName);
+  const [nickname, setNickname] = useState(profileData?.profileInfo.nickname);
+  const [profileImgUrl, setProfileImgUrl] = useState(profileData?.profileInfo.profileImage);
+
+  // Note: initial getItem from localStorage after browser refresh will error
+  // since the old file blob will not longer be valid, however, because
+  // profileImgUrl would be reset, a new profileImageBlob would be set from
+  // handleGetProfileImage
+  const [profileImg, setProfileImg] = useState(
+    JSON.parse(localStorage.getItem('profileImageBlob'))
+  );
+
+  const [edit, setEdit] = useState(false);
+
+  const loadProfileData = async () => {
+    const profileDataSolid = await fetchProfileInfo(session);
+    localStorage.setItem('restoreProfileData', JSON.stringify(profileDataSolid));
```

Comment:
`localStorage` usually has a maximum size of 5mb. This will error on any image larger than that (and there are a lot of images larger than 5mb). You don't need to handle loading the image anyway. Just let the browser fetch it.

Reply:
I think this is fine. What is being stored is just the string of the link to the blob like `blob:http://localhost:5173/5d4def0c-28eb-4d7d-913d-46a92f23b991` not the actual image file.

Reply:
Then where are you storing the blob?

Reply:
I'm using the `createObjectURL` from the `URL` API to reference the image file blob. So the blob is consumed to generate a reference URL to the image file which is stored in memory on the browser.

I think we could include an additional `URL.revokeObjectURL` function to explicitly release them from memory.

Reply:
That describes how you create it, but not where the data is stored. It seems to end up in browser memory, which is normally cleared when the browser closes or the system restarts. There may be a bit more persistence, but it will vary between browsers, versions, and OSes. Either way I don't think this is a reliable solution. If you pass the URL to the `src` attribute of the `img` tag, you don't need any of this load handling, and it will perform faster.

Reply:
I'm not too certain on an alternative at this time. I could take out the localStorage links to the reference URL, but the result would be it'll have to load the blob each time. See clip.

https://github.com/codeforpdx/PASS/assets/14917816/152ce169-60b6-40b3-8840-157f1aaeafe9

With regards to browsers (at least with the ones I've tested with) it seems to be consistent using this approach.

https://github.com/codeforpdx/PASS/assets/14917816/91d16ad5-5fcb-4acd-9282-98e6f047dd61

For the URL, I could only pass in the reference URL after the blob object from Solid is referenced.

Reply:
Oh! I get it. I'll test it out with just the URL link to the picture on Solid.

Reply:
I'm not really sure what I'm doing wrong, but I can't seem to get the flickering behavior to go away even when using the direct URL to the image file which I find surprising. The reference URL from the blob seems to work okay though.

https://github.com/codeforpdx/PASS/assets/14917816/1d8e0583-a230-4ab3-bd54-2bf5ab8e1246

I didn't expect the direct link to the image would force a re-render of the image, the `profileImg` prop just passes a URL to `src` in `Avatar` without any processing in Line 71.

![Screen Shot 2023-07-04 at 14 22 53](https://github.com/codeforpdx/PASS/assets/14917816/4ba2a04c-b87e-4ec5-b1bd-51358a7a0605)

Reply:
Think I've got it to work for the most part. It's been chipped own to just use just the url to the image file on Solid rather than the blob.

Still using `localStorage` for persistence, but it's all moved to inside `ProfileImageField` and I've simplified to a single `useState` (link to image file on Solid) rather than two (one for the blob, one for link to image file on Solid).

`Navbar` and `NavMenu` simply pulls from `localStorage` to render the profile image.

Reply:
Think I got the flickering disappear as well. Think this solution would be good for our "Profile" page.

https://github.com/codeforpdx/PASS/assets/14917816/10509e94-644f-4480-860e-04b5e67fb9e5